### PR TITLE
Fix broken `point_indices` methods in `PolygonGeoColumnAccessor`

### LINKED ISSUE #891: [BUG]: `GeoSeries.polygons.point_indices()` throws error
### Version

23.04

### On which installation method(s) does this occur?

Rapids-Compose

### Describe the issue

`GeoSeries.polygons.point_indices()` returns a mapping from the individual points of a polygon to the part numbers that they are from. This method throws an error currently and needs to be updated, or replaced, to support `.contains_properly` being based on quadtree pip.

### Minimum reproducible example

```shell
import geopandas as gpd
import cuspatial
dataframe = cuspatial.GeoDataFrame(gpd.read_file(
    gpd.datasets.get_path("naturalearth_lowres")
))
dataframe['geometry'].polygons.point_indices()
```

### Relevant log output

```shell
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In[27], line 7
      3 import cuspatial
      4 dataframe = cuspatial.GeoDataFrame(gpd.read_file(
      5     gpd.datasets.get_path("naturalearth_lowres")
      6 ))
----> 7 dataframe['geometry'].polygons.point_indices()

File ~/cuspatial/python/cuspatial/cuspatial/core/geoseries.py:242, in GeoSeries.PolygonGeoColumnAccessor.point_indices(self)
    240 offsets = cp.array(self.ring_offset)
    241 sizes = offsets[1:] - offsets[:-1]
--> 242 return cp.repeat(self._series.index, sizes)

File ~/compose/etc/conda/cuda_11.6/envs/notebooks/lib/python3.8/site-packages/cupy/_manipulation/tiling.py:70, in repeat(a, repeats, axis)
     56 def repeat(a, repeats, axis=None):
     57     """Repeat arrays along an axis.
     58 
     59     Args:
   (...)
     68 
     69     """
---> 70     return a.repeat(repeats, axis)

File ~/cudf/python/cudf/cudf/core/index.py:809, in RangeIndex.repeat(self, repeats, axis)
    808 def repeat(self, repeats, axis=None):
--> 809     return self._as_int_index().repeat(repeats, axis)

File ~/cudf/python/cudf/cudf/core/index.py:1492, in GenericIndex.repeat(self, repeats, axis)
   1490 def repeat(self, repeats, axis=None):
   1491     return self._from_columns_like_self(
-> 1492         Frame._repeat([*self._columns], repeats, axis), self._column_names
   1493     )

File ~/cudf/python/cudf/cudf/core/frame.py:2810, in Frame._repeat(columns, repeats, axis)
   2807 if not is_scalar(repeats):
   2808     repeats = as_column(repeats)
-> 2810 return libcudf.filling.repeat(columns, repeats)

File ~/compose/etc/conda/cuda_11.6/envs/notebooks/lib/python3.8/contextlib.py:75, in ContextDecorator.__call__.<locals>.inner(*args, **kwds)
     72 @wraps(func)
     73 def inner(*args, **kwds):
     74     with self._recreate_cm():
---> 75         return func(*args, **kwds)

File filling.pyx:57, in cudf._lib.filling.repeat()

File filling.pyx:68, in cudf._lib.filling._repeat_via_column()

RuntimeError: CUDF failure at: /home/tcomer/mnt/NVIDIA/rapids-docker/cudf/cpp/src/filling/repeat.cu:109: in and count must have equal size
```

### Environment details

```shell
rapids-compose 23.04
```

### Other/Misc.

_No response_

### PR DESCRIPTION
**Pull request by isVoid**
## Description

Closes #891 , fixes an issue in `point_indices`. Previously, the point indices are computed with the ring offset, but the target index is mean to denote the (multi)polygon index, so we need to compute an array of offset corresponding to geometry offset. This can be computed via nested gathers.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by thomcom** [comment]
I too have written a solution for this that appears much more complicated than yours. We should probably put our heads together and make sure that we're doing the same thing, and if so I'm unsurprised that yours is much more parsimonious. :)

File: python/cuspatial/cuspatial/tests/test_geoseries.py [hunk 1]
```diff
@@ -621,3 +622,97 @@ def test_reset_index(level, drop, name, inplace):
         pd.testing.assert_series_equal(expected, got.to_pandas())
     else:
         pd.testing.assert_frame_equal(expected, got.to_pandas())
+
+
+def test_geocolumn_polygon_accessor(naturalearth_lowres):
```
Comment (thomcom) [inline python/cuspatial/cuspatial/tests/test_geoseries.py:627]: Looks like you forgot to add this fixture.

**Review by thomcom — CHANGES_REQUESTED** [review]

File: python/cuspatial/cuspatial/tests/test_geoseries.py (hunk 1 above)
Comment (isVoid) [reply python/cuspatial/cuspatial/tests/test_geoseries.py]: Removed, this is not needed.

**Comment by isVoid** [comment]
> I too have written a solution for this that appears much more complicated than yours. We should probably put our heads together and make sure that we're doing the same thing, and if so I'm unsurprised that yours is much more parsimonious. :)

Commenting on admin side of things, given the asynchronous nature of how the team is organized, I think what is really missing is how to better communicate what tickets one is currently working on so it's not being double worked on. This can be done easily by clicking `assign yourself` on the issue ticket.

If you have taken up the issue ticket for this PR: https://github.com/rapidsai/cuspatial/issues/891, please click assign so it's not worked on twice.

**Comment by thomcom** [comment]
My apologies, I ended up writing it up with the `.contains_properly` refactor to use quadtree as it was required there. Please freeze your work on this for now and we can discuss soon.

File: python/cuspatial/cuspatial/core/geoseries.py [hunk 2]
```diff
@@ -209,7 +209,8 @@ def part_offset(self):
         def point_indices(self):
             # Return a cupy.ndarray containing the index values from the
             # LineString GeoSeries that each individual point is member of.
-            offsets = cp.array(self.part_offset)
+            # offsets = cp.array(self.part_offset)
```
Comment (isVoid) [inline python/cuspatial/cuspatial/core/geoseries.py:212]:
```suggestion
```

**Review by isVoid — COMMENTED** [review]
stale comment

**Comment by thomcom** [comment]
Yup, seems to work as well as mine and is as expected much more parsimonious.

**Review by thomcom — APPROVED** [review]

**Pull request merged**